log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java

```diff
@@ -148,7 +148,8 @@ public static RegexFilter createFilter(
             LOGGER.error("A regular expression must be provided for RegexFilter");
             return null;
         }
-        return new RegexFilter(useRawMsg, Pattern.compile(regex, toPatternFlags(patternFlags)), match, mismatch);
+        return new RegexFilter(
+                useRawMsg != null && useRawMsg, Pattern.compile(regex, toPatternFlags(patternFlags)), match, mismatch);
```

Comment:
_Nit_: Maybe comparing `useRawMsg` to `Boolean.TRUE` might be simpler to read:
```suggestion
        return new RegexFilter(
                Boolean.TRUE.equals(useRawMsg), Pattern.compile(regex, toPatternFlags(patternFlags)), match, mismatch);
```

Reply:
+1, yeah I like the suggestion, let us do that!